tests/test_basic_functionality.py

```diff
@@ -343,6 +344,36 @@ def test_dictionary_style_access() -> None:
     assert lct["test_key"] == "test_value"
 
 
+def test_dunder_key_attr_style_access() -> None:
+    lct = LayeredConfigTree()
+    lct.update({"__dunder_key__": "val"})
+    assert lct["__dunder_key__"] == "val"
+
+    with pytest.raises(
+        RuntimeError,
+        match=re.escape(
+            "Cannot get an attribute starting and ending with '__' via attribute "
```

Comment:
This is a nit, but I generally prefer using a key word from the error message rather than the whole thing, to allow wording changes in the future without test diff. No idea what the best practice is!

Reply:
Interesting, b/c I prefer entire messages myself so we don't accidentally pass. I can see it going either way